src/ConnectionTransports/NetworkSocketConnectionTransport.cpp

```diff
@@ -25,335 +25,222 @@ NetworkSocketConnectionTransport::NetworkSocketConnectionTransport(
-            datagramsPendingWrite.push_back(bytes);
-            // write a single byte to indicate that we've got pending data
-            const char buf[] = { 0 };
-            writeResult = write(writePipeFds[1], &buf[0], 1);
-            expectedBytesWritten = 1;
-        }
-        else
-        {
-            writeResult = write(writePipeFds[1], bytes.data(), bytes.size());
-            expectedBytesWritten = bytes.size();
-        }
-        
-        if (writeResult != expectedBytesWritten)
-        {
-            spdlog::error(
-                "Write returned {} result, expected {} bytes.",
-                writeResult,
-                bytes.size());
-            stoppingLock.unlock(); // Unlock so we can start stopping :)
-            closeConnection();
-        }
-    }
+    return targetAddr;
 }
 
-void NetworkSocketConnectionTransport::SetOnConnectionClosed(
-    std::function<void(void)> onConnectionClosed)
+std::optional<sockaddr_in6> NetworkSocketConnectionTransport::GetAddr6()
 {
-    this->onConnectionClosed = onConnectionClosed;
+    // TODO
+    return std::nullopt;
 }
 
-void NetworkSocketConnectionTransport::SetOnBytesReceived(
-    std::function<void(const std::vector<std::byte>&)> onBytesReceived)
+Result<ssize_t> NetworkSocketConnectionTransport::Read(std::vector<std::byte>& buffer)
 {
-    this->onBytesReceived = onBytesReceived;
-}
-#pragma endregion ConnectionTransport Implementation
+    std::scoped_lock lock(readMutex);
 
-#pragma region Private methods
-void NetworkSocketConnectionTransport::connectionThreadBody(
-    std::promise<void>&& connectionThreadEndedPromise)
-{
-    // Indicate when this thread has exited.
-    connectionThreadEndedPromise.set_value_at_thread_exit();
+    if (isStopped)
+    {
+        return Result<ssize_t>::Error("Transport is stopped");
+    }
 
-    // We're connected. Now wait for input/output.
-    char readBuf[BUFFER_SIZE] = { 0 };
-    while (true)
+    pollfd pollFds[]
     {
-        pollfd pollFds[]
+        // Socket read
         {
-            // Socket read
+            .fd = socketHandle,
+            .events = POLLIN,
+            .revents = 0,
+        },
+    };
+
+    poll(pollFds, 1, READ_TIMEOUT_MS);
+
+    // Did the socket get closed?
+    if (((pollFds[0].revents & POLLERR) > 0) || 
+        ((pollFds[0].revents & POLLHUP) > 0) ||
+        ((pollFds[0].revents & POLLNVAL) > 0))
+    {
+        return Result<ssize_t>::Error("Socket closed");
+    }
+
+    // Data available for reading?
+    if (pollFds[0].revents & POLLIN)
+    {
+        sockaddr_in recvFromAddr{};
+        socklen_t recvFromAddrLen = sizeof(recvFromAddr);
+
+        buffer.resize(BUFFER_SIZE);
+        ssize_t bytesRead = recvfrom(
+            socketHandle,
+            buffer.data(),
+            buffer.size(),
+            0,
+            reinterpret_cast<sockaddr*>(&recvFromAddr),
+            &recvFromAddrLen);
+
+        if (bytesRead == -1)
+        {
+            buffer.resize(0);
```

Comment:
In some of the other error cases we don't blow away the buffer - we should decide whether we want to blow it away for all error cases, or leave it untouched.

Reply:
Updated to always size the buffer before returning.